### DISCUSSION TITLE
视频文件调用 MXPlayer, MXPlayerPro, nPlayer

### DISCUSSION THREAD

**Discussion by gdm257** [AlistGo/alist#605] [Ideas]
如题，现有播放器不方便手机观看，如果能多加几个播放器就全平台通杀了

例如 alist 视频文件下载链接为 `http://abc.com/d/movie/video.mkv`

1. 调用 MXPlayer (free)
`intent:http://abc.com/d/movie/video.mkv#Intent;package=com.mxtech.videoplayer.ad;S.title=undefined;end`
`intent:<文件URL>#Intent;package=com.mxtech.videoplayer.ad;S.title=undefined;end`

2. 调用 MXPlayer Pro
`intent:http://abc.com/d/movie/video.mkv#Intent;package=com.mxtech.videoplayer.pro;S.title=undefined;end`
`intent:<文件URL>#Intent;package=com.mxtech.videoplayer.pro;S.title=undefined;end`

3. 调用 nPlayer
`nplayer-http://abc.com/d/movie/video.mkv`
`nplayer-<文件URL>`

**Comment by ErDongChen1973** [comment]
> 如题，现有播放器不方便手机观看，如果能多加几个播放器就全平台通杀了
> 
> 例如 alist 视频文件下载链接为 `http://abc.com/d/movie/video.mkv`
> 
> 1. 调用 MXPlayer (free)
> `intent:http://abc.com/d/movie/video.mkv#Intent;package=com.mxtech.videoplayer.ad;S.title=undefined;end`
> `intent:<文件URL>#Intent;package=com.mxtech.videoplayer.ad;S.title=undefined;end`
> 
> 2. 调用 MXPlayer Pro
> `intent:http://abc.com/d/movie/video.mkv#Intent;package=com.mxtech.videoplayer.pro;S.title=undefined;end`
> `intent:<文件URL>#Intent;package=com.mxtech.videoplayer.pro;S.title=undefined;end`
> 
> 3. 调用 nPlayer
> `nplayer-http://abc.com/d/movie/video.mkv`
> `nplayer-<文件URL>`
> 
> 
> 

这个是怎么用呢？

**Reply by gdm257** [reply]
> > 如题，现有播放器不方便手机观看，如果能多加几个播放器就全平台通杀了
> > 例如 alist 视频文件下载链接为 `http://abc.com/d/movie/video.mkv`
> > 
> > 1. 调用 MXPlayer (free)
> >    `intent:http://abc.com/d/movie/video.mkv#Intent;package=com.mxtech.videoplayer.ad;S.title=undefined;end`
> >    `intent:<文件URL>#Intent;package=com.mxtech.videoplayer.ad;S.title=undefined;end`
> > 2. 调用 MXPlayer Pro
> >    `intent:http://abc.com/d/movie/video.mkv#Intent;package=com.mxtech.videoplayer.pro;S.title=undefined;end`
> >    `intent:<文件URL>#Intent;package=com.mxtech.videoplayer.pro;S.title=undefined;end`
> > 3. 调用 nPlayer
> >    `nplayer-http://abc.com/d/movie/video.mkv`
> >    `nplayer-<文件URL>`
> 
> 这个是怎么用呢？

可以参考 alist 已支持的 potplayer IINA VLC。本质都是 URL，用浏览器打开这些链接，就能调用对应播放器（alist 的按钮实际是模拟这一行为），前提是你安装了对应播放器
